### DISCUSSION TITLE
Discussion to Move up the Minimum Qt version PyQtGraph supports

### DISCUSSION THREAD

**Discussion by j9ac9k** [pyqtgraph/pyqtgraph#2332] [Polls]
Through a lot of hard work, there are now conda packages for PyQt5 and PySide2 on the conda-forge channel for Qt5 5.15.4+

* [PyQt package](https://anaconda.org/conda-forge/pyqt)
* [PySide package](https://anaconda.org/conda-forge/pyside2)

We have largely kept Qt 5.12 support alive because the newest conda package available for python bindings used Qt 5.12, but now that there are packages for 5.15; I'm inclined to remove pyqtgraph support for anything older than 5.15.

Before doing that, I would like to get some community input.  If you need support for something older than 5.15, I would like to know your use case.

Moving to a newer Qt form won't have any immediate effects; there would be some if-checks/workarounds we can get rid of, and our CI pipelines can be a bit more targeted as well.  We have generally found that removing support for older versions of Qt doesn't have major initial benefits, but does free up developer time to not have to try and accommodate environments that are likely not used much.

**Comment by mliberty1** [comment]
For what it's worth, I am planning on migrating the [Joulescope UI](https://github.com/jetperch/pyjoulescope_ui) to PySide6 this summer.  I would be fine if PyQtGraph dropped PySide2 / PyQt5 support entirely.

**Reply by j9ac9k** [reply]
Hi @mliberty1,

Good to hear from you.  Dropping Qt5 support definitely won't be happening for a while, there was a [new PyQt5 release yesterday actually](https://pypi.org/project/PyQt5/#history).  Qt6 changeover hasn't been that big of a jump, so I don't anticipate the change being that disruptive for Joulscope.  We did run into a major issue with PySide6 6.3.0 on Python 3.10, but that's been resolved.  The PySide devs are working on adding PyPy support; we tried their first go of that but found a number of show-stopper issues (unless you're using `QOpenGLWidget`).

**Reply by mliberty1** [reply]
In case you haven't seen [NEP 29 — Recommend Python and NumPy version support as a community policy standard](https://numpy.org/neps/nep-0029-deprecation_policy.html), check it out.  PyQtGraph could just adopt this policy.  Since PySide2 5.15.0 was released May 27, 2020 (see [history](https://pypi.org/project/PySide2/#history)), you are good to drop support for 5.15.0 and older.

In other news, I have been off working on a new product and not doing any new work with PyQtGraph.  I will be getting back to development with PyQtGraph this summer, and I look forward to checking out the new features.  I'll try out new features (2) and (3) from your [post](https://groups.google.com/g/pyqtgraph/c/UHb4TR4ly4M/m/HjkecayMAwAJ), hopefully, late Jul / early Aug.  Thank you for your continued work on PyQtGraph!

**Reply by j9ac9k** [reply]
We adopted NEP-29 quite some time ago actually, it's made ongoing development and maintenance so much easier! I did consider applying that kind of fixed timeline of dropping support for Qt versions, but I decided it wouldn't work as Qt bindings on conda were not made available at the same rate they were made available as pip wheels.

Most of PyQtGraph's new work has been in performance improvements, which there have been substantial ones made with almost all plot types.

**Comment by markotoplak** [comment]
Not directly related to pyqtgraph, but a fellow dev perspective. Our package (Orange) also required some workarounds for PyQt 5.12. Almost as soon as PyQt 5.15 was available on conda-forge we dropped support for 5.12 and were happy to do so.

**Reply by j9ac9k** [reply]
Hey @markotoplak good to hear from you.  Glad you all were able to make this change without issue.

**Comment by j9ac9k** [comment]
Going to lock this as with PyQtGraph 0.13.0 we now mandate Qt 5.15+ for Qt5.  Further good news on the subject is that the conda defaults channel also has Qt 5.15 in there too now, so conda users won't be left so far behind.
